server/app/uk/gov/ons/addressIndex/server/controllers/EQController.scala

```diff
@@ -29,18 +30,82 @@ class EQController @Inject () (val controllerComponents: ControllerComponents,
               favourwelsh: Option[String] = None
              ): Action[AnyContent] = Action async { implicit req =>
 
-    if (isPostCode(input)) {
+  // we want to test the string with spaces as supplied by the user first
+  // if no match try again with spaces removed.
+    val normalizedInput: String = stripSpaces(input.toUpperCase)
+
+    if (isPostCode(normalizedInput)) {
       logger.info("Input is postcode")
       eqPostcodeController.postcodeQuery(
-        postcode = input,
+        postcode = normalizedInput,
         offset = offset,
         limit = limit,
         classificationfilter = classificationfilter,
         historical = historical,
         verbose = verbose,
         favourpaf = favourpaf,
         favourwelsh = favourwelsh,
-        epoch=epoch) (req)
+        epoch = epoch)(req)
+    } else if (isOutCodeAndSectorAndHalfUnitWithSpace(input)){
```

Comment:
Looks like this is in need of some refactoring?

Reply:
Yes. Separate functions and mutilple elseif blocks do allow the logic and preference order to be implemented as described in the notes on Confluence (and easily modified), but it can ultimately all be condensed into a single match function for neatness.